### PR TITLE
usb: Update to use Tock2.0 Driver

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

Update the USB Tock2.0 Driver to use the new `Driver` API.

### Testing Strategy

I don't have a way to test this unfortunately. The changes seem very simple so I think it's fine.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: kernel/src/callback.rs [hunk 1]
```diff
@@ -196,6 +196,10 @@ impl Callback {
         self.cb.map_or(true, |mut cb| cb.schedule(r0, r1, r2))
     }
 
+    pub fn is_some(&self) -> bool {
+        self.cb.is_some()
+    }
+
```
Comment (hudson-ayers) [inline kernel/src/callback.rs:202]: I believe this is something we explicitly do not want to provide. See the discussion here: https://github.com/tock/tock/pull/2258#discussion_r542765769

**Review by hudson-ayers — CHANGES_REQUESTED** [review]

**Review by hudson-ayers — APPROVED** [review]

**Comment by phil-levis** [comment]
Needs testing.

**Comment by phil-levis** [comment]
Discussion on the core call today reached consensus that it's OK to merge this untested with the -dev branch, but will need testing before merging -dev into master.

**Pull request merged**